### PR TITLE
2215 issue tracker

### LINKED ISSUE #2215: Allow for adding links to Jira/Azure DevOps/GitLab/Trello issues including status of the issue
This makes it possible to see in Quality-time what the status of a linked issue us.

UI:
- Allow for configuring multiple issue trackers per Quality-time instance, on the home page. Per issue tracker: Issue tracker type (dropdown with issue tracker types: Jira/Azure DevOps/GitLab/Trello) + Name + URL + credentials.
- Add input fields to metric details for entering an issue: Issue tracker (dropdown with configured issue trackers) + issue id.
- Add column to metric table with "Linked issues". Text of the issue is the issue id. The issue links to the issue in the issue tracker. The status is shown as: 
```jsx
    <Label as='a' color='blue'>
      <a href="http://issue_tracker.example.org/issue-id-123">ISSUE-ID-123</a>
      <Label.Detail>Open</Label.Detail>
    </Label>
```
- The UI loads the statuses of the metrics after (re)loading the report, by asking for the statuses of the issues in the report. 

Server:
- Add API endpoint for synchronously requesting the status of an issue for a specific report uuid.
- Add module for talking to issue trackers, with submodules per issue tracker type.

Data model:
- For each source that is also an issue tracker add a `issue_tracker` object:
```json
  "jira": {
    "name": "Jira",
    "parameters": {},
    "issue_tracker": {
      "help": "How to find the issue id: ..."
     } 
  }
```

Slices:
- Jira first.

### PR DESCRIPTION
**Pull request by emdv88**
Closes #2215.

### CODE REVIEW

File: components/frontend/src/App.js [hunk 1]
```diff
@@ -265,11 +265,14 @@ class App extends Component {
...
-                  {...props}
-                />
+                <>
+                  <Reports
+                    datamodel={this.state.datamodel}
+                    open_report={(e, r) => this.open_report(e, r)}
+                    reports_overview={this.state.reports_overview}
+                    {...props}
+                  />
+                </>
```
Comment (fniessink) [inline components/frontend/src/App.js]: Why is `<Reports>` embedded in a fragment?

File: components/frontend/src/App.js [hunk 2]
```diff
@@ -123,7 +123,7 @@ class App extends Component {
             datamodel: data_model,
             reports: reports.reports || [],
             reports_overview: {
-              layout: reports.layout, subtitle: reports.subtitle, title: reports.title, permissions: reports.permissions },
+              layout: reports.layout, subtitle: reports.subtitle, title: reports.title, permissions: reports.permissions, issueTrackers: reports.issue_trackers || {} },
```
Comment (fniessink) [inline components/frontend/src/App.js]: I'd prefer consistent casing...

**Review by fniessink — COMMENTED** [review]
If it's not too much trouble it would be nice to move the expansion of the props parameter to a different PR as it touches a lot of files.

File: components/frontend/src/App.js (hunk 2 above)
Comment (emdv88) [reply components/frontend/src/App.js]: I agree.. but then I'm wondering where to insert empty dictionaries if nothing is present. Maybe the best solution is during the database initialization to fill it in all old report_overviews.

File: components/frontend/src/App.js (hunk 1 above)
Comment (emdv88) [reply components/frontend/src/App.js]: mh good question...

File: components/frontend/src/App.js (hunk 2 above)
Comment (fniessink) [reply components/frontend/src/App.js]: Agreed.

**Review by fniessink — COMMENTED** [review]
Open issues:

- [ ] The popup of an issue with a connection error is empty.
- [x] "Issuetracker" should be spelled "issue tracker".
- [ ] The column header "Linked Issue" should be spelled "Linked issue".
- [ ] The tracker password is visible in the change log.
- [ ] Time travel shows the current status of the issue, not the status at the date traveled to: consider making the collector responsible for collecting the issue status. Would reduce duplication between server and collector.

**Pull request merged**